lib/install/management/delete.go

```diff
@@ -178,6 +178,43 @@ func (d *Dispatcher) DeleteVCHInstances(vmm *vm.VirtualMachine, conf *config.Vir
 			continue
 		}
 		if ok {
+			// child is VCH -- detach container disk images to prevent undeletable state
+			devices, err := child.Device(d.ctx)
+			if err != nil {
+				log.Errorf("error getting devices %s:", err.Error())
+				continue
+			}
+
+			disks := devices.SelectByType(&types.VirtualDisk{})
+			if disks == nil {
+				continue
+			}
+
+			for _, disk := range disks {
+				config := []types.BaseVirtualDeviceConfigSpec{
+					&types.VirtualDeviceConfigSpec{
+						Device:    disk,
+						Operation: types.VirtualDeviceConfigSpecOperationRemove,
+					},
+				}
+				spec := types.VirtualMachineConfigSpec{}
+				spec.DeviceChange = config
+
+				op := trace.NewOperation(d.ctx, "detach disks before delete")
+				_, err := child.WaitForResult(op,
```

Comment:
cause you need to detach all disk devices and it's possible to have multiple devices attached at this time, it would be better to generate vm reconfigure spec inside of for loop, and reconfigure vm just once.
btw, there is one method in govmomi VirtualMachine object, called `RemoveDevice`, which could remove devices directly, so you don't need to create vm reconfigure spec and call task.wait...